File: algorithms/187. Repeated DNA Sequences/solution.cc
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        // @1337c0d3r
        unordered_map<int, int> counter;
        vector<string> result;
        for ( int t = 0, i = 0, n = s.size(); i < n; ++i )
            if ( ++counter[t = t << 3 & 0x3FFFFFFF | s[i] & 7] == 2 )
                result.push_back(s.substr(i - 9, 10));
        return result;
    }
};

// 32 / 32 test cases passed.
// Status: Accepted
// Runtime: 56 ms
// beats 72.83 %

File: algorithms/187. Repeated DNA Sequences/solution.py
class Solution(object):
    def findRepeatedDnaSequences(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        counter = collections.defaultdict(int)
        for i in xrange(10, len(s) + 1):
            counter[s[i-10:i]] += 1
        return [ss for ss in counter if counter[ss] > 1]

# 32 / 32 test cases passed.
# Status: Accepted
# Runtime: 155 ms
# beats 63.38 %
